lib/api/core/plugins/pluginsManager.js

```diff
@@ -751,20 +785,223 @@ function registerHook(kuzzle, plugin, event, fn) {
   });
 }
 
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {object} authentications
+ * @param {object} plugin
+ * @param {string} pluginName
+ */
+function injectAuthentication(kuzzle, authentications, plugin, pluginName) {
+  const
+    mandatoryMethods = ['exists', 'create', 'update', 'delete', 'validate', 'verify'],
+    errorPrefix = `[Plugin Manager] Error initializing plugin ${pluginName}:`;
+  let valid = true;
+
+  if (!isObject(plugin.strategies) || Object.keys(plugin.strategies).length === 0) {
+    // eslint-disable-next-line no-console
+    console.error(`${errorPrefix} The plugin must provide an object "strategies".`);
+    valid = false;
+
+    return;
+  }
+
+  Object.keys(plugin.strategies).forEach(strategyName => {
+    /** @type AuthenticationStrategy */
+    const strategy = plugin.strategies[strategyName];
+    if (!isObject(strategy)) {
+      // eslint-disable-next-line no-console
+      console.error(`${errorPrefix} The plugin must provide an object for strategy "${strategyName}".`);
+      valid = false;
+
+      return;
+    }
+
+    if (kuzzle.pluginsManager.registeredStrategies.indexOf(strategyName) !== -1) {
+      // eslint-disable-next-line no-console
+      console.error(`[Plugin Manager] An authentication strategy "${strategyName}" has already been registered.`);
+      throw new PluginImplementationError(`[Plugin Manager] An authentication strategy "${strategyName}" has already been registered.`);
+    }
+
+    if (!isObject(strategy.methods)) {
+      // eslint-disable-next-line no-console
+      console.error(`${errorPrefix} The plugin must provide a "methods" object in strategies['${strategyName}'] property.`);
+      valid = false;
+    }
+    else {
+      mandatoryMethods.forEach(methodName => {
```

Comment:
OK, I have to leave a comment despite the counter-arguments listed below:

All methods could be declared as first-citizen functions. We don't need to use a string representation of the property to address.

### counter arguments

1. This is how hooks, pipes and controllers work

Changing it now has too much impact just before a major release

2. This impacts plugin developers only

We can expect this audience to be more kind to adapt to some little oddities like this than for the front-end API

#### Actions

imo, to be tackled after 1.0 
https://github.com/kuzzleio/kuzzle/issues/834